src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc

```diff
@@ -141,83 +145,93 @@ PluggableAuthExternalAccountCredentials::
     output_file_path_ = getStringValue(executable_json, "output_file");
 }
 
-void PluggableAuthExternalAccountCredentials::CreateExecutableResponse(
-    std::string executable_output_string) {
+PluggableAuthExternalAccountCredentials::ExecutableResponse*
+PluggableAuthExternalAccountCredentials::CreateExecutableResponse(
+    std::string executable_output_string, grpc_error_handle* error) {
   auto executable_output = JsonParse(executable_output_string);
+  ExecutableResponse* executable_response =
+      (ExecutableResponse*)gpr_malloc(sizeof(ExecutableResponse));
   if (!executable_output.ok()) {
-    FinishRetrieveSubjectToken(
-        "", GRPC_ERROR_CREATE(
-                "The response from the executable contains an invalid "
-                "or malformed response."));
+    *error = GRPC_ERROR_CREATE(
+        "The response from the executable contains an invalid "
+        "or malformed response.");
+    return nullptr;
   }
   if (!isKeyPresent(executable_output, "version")) {
-    FinishRetrieveSubjectToken(
-        "", GRPC_ERROR_CREATE("The executable response must contain the "
-                              "`version` field."));
+    *error = GRPC_ERROR_CREATE(
+        "The executable response must contain the "
+        "`version` field.");
+    return nullptr;
   }
   absl::SimpleAtoi(getStringValue(executable_output, "version"),
-                   &executable_response_.version);
+                   &executable_response->version);
   auto executable_output_it = executable_output->object().find("success");
   if (!isKeyPresent(executable_output, "success")) {
-    FinishRetrieveSubjectToken(
-        "", GRPC_ERROR_CREATE("The executable response must contain the "
-                              "`success` field."));
+    *error = GRPC_ERROR_CREATE(
+        "The executable response must contain the "
+        "`success` field.");
+    return nullptr;
   }
-  executable_response_.success = executable_output_it->second.boolean();
-  if (executable_response_.success) {
+  executable_response->success = executable_output_it->second.boolean();
+  if (executable_response->success) {
     if (!isKeyPresent(executable_output, "token_type")) {
-      FinishRetrieveSubjectToken(
-          "", GRPC_ERROR_CREATE("The executable response must contain the "
-                                "`token_type` field."));
+      *error = GRPC_ERROR_CREATE(
+          "The executable response must contain the "
+          "`token_type` field.");
+      return nullptr;
     }
-    executable_response_.token_type =
-        getStringValue(executable_output, "token_type");
-    executable_response_.expiration_time = 0;
+    executable_response->token_type =
+        gpr_strdup(getStringValue(executable_output, "token_type").c_str());
+    executable_response->expiration_time = 0;
```

Comment:
Shouldn't expiration time just not get set rather than getting set to 0 here?

Reply:
expiration_time will have junk value if it's uninitialized. I am going to set it to "gpr_inf_future" time, since we assume if the expiration_time field is not present, the response from an executable is unexpired by default (if output file is not specified). 
There is a check that throws an error when expiration_time is not present and when output_file is specified.